packages/theme/pages/Product.vue

```diff
@@ -470,61 +462,76 @@ export default defineComponent({
       });
     };
 
-    const updateProductConfiguration = async (label, value) => {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      productConfiguration.value.push([label, value]);
+    let firstFetch = true;
+    const { fetch } = useFetch(async () => {
+      const { params: { id } } = route.value;
 
-      await router.push({
-        path: `${app.localePath(route.value.fullPath)}`,
-        query: {
-          ...Object.fromEntries(productConfiguration.value),
-        },
-      });
-    };
-
-    useAsync(async () => {
       const baseSearchQuery = {
         filter: {
           sku: {
             eq: id,
           },
         },
-        ...(productConfiguration.value.length > 0
-          ? {
-            configurations: productConfiguration.value.map(
-              (config) => config[1],
-            ),
-          }
-          : {}),
+        configurations: productConfiguration.value.map(
+          (config) => config[1],
+        ) as string[],
       };
 
-      products.value = await getProductDetails({
+      const result = await getProductDetails({
         ...baseSearchQuery,
       });
 
-      if (product?.value?.length === 0) nuxtError({ statusCode: 404 });
+      product.value = result.items[0] as Product ?? null;
+      if (Boolean(product?.value?.sku) === false) nuxtError({ statusCode: 404 });
```

Comment:
```suggestion
      if (!Boolean(product?.value?.sku)) nuxtError({ statusCode: 404 });
```
nitpick, can commit through github